db/seeds.rb

```diff
@@ -17,171 +17,307 @@
-    dimensions: [50, 20]
-  )
-
-  Request.first.update!(status: :accepted)
-
-  RequestDistribution.create!(
-    request: Request.second,
-    bench: Bench.second,
-    plant_stage: Request.second.plant_stage,
-    pot: Pot.second,
-    pot_quantity: 20,
-    positions_on_bench: [0, 10],
-    dimensions: [20, 30]
-  )
+  requests = []
+  5.times do |i|
+    plant = Plant.order('RANDOM()').first
+    plant_stage = plant.plant_stages.order('RANDOM()').first
+    requests << Request.create!(
+      requester_first_name: Faker::Name.first_name,
+      requester_last_name: Faker::Name.last_name,
+      requester_email: Faker::Internet.email,
+      laboratory: Faker::Science.scientist,
+      handler: User.first,
+      plant_stage: plant_stage,
+      name: "Requête #{i + 1}",
+      plant_name: plant.name,
+      plant_stage_name: plant_stage.name,
+      comment: Faker::Lorem.sentence,
+      due_date: Date.current + rand(1..6).months,
+      quantity: rand(10..100),
+      temperature: %w[Chaud Froid Extérieur].sample,
+      photoperiod: rand(6..18),
+      status: %i[pending refused completed].sample
+    )
+  end
+
+  def distribute_request(request)
+    return unless request_pending_or_accepted?(request)
+
+    bench, pot = select_random_bench_and_pot
+    distribution_details = prepare_distribution_details(request, bench)
+
+    handle_distribution(
+      request: request,
+      bench: bench,
+      pot: pot,
+      distribution_details: distribution_details
+    )
+  end
+
+  def request_pending_or_accepted?(request)
+    %w[pending accepted].include?(request.status)
+  end
+
+  def select_random_bench_and_pot
+    [Bench.order('RANDOM()').first, Pot.order('RANDOM()').first]
+  end
+
+  def prepare_distribution_details(request, bench)
+    dimensions = [rand(50..200), rand(50..200)]
+    position = find_position_for_request(bench, dimensions)
+    distributed_quantity = rand(1..request.quantity)
+
+    {
+      dimensions: dimensions,
+      position: position,
+      quantity: distributed_quantity,
+      total_quantity: request.quantity
+    }
+  end
+
+  def find_position_for_request(bench, dimensions)
+    occupied_positions = []
+    find_available_position(bench, dimensions, occupied_positions)
+  end
+
+  def handle_distribution(request:, bench:, pot:, distribution_details:)
+    if distribution_details[:position]
+      create_distribution_record(
+        request: request,
+        bench: bench,
+        pot: pot,
+        details: distribution_details
+      )
+    end
+
+    update_request_status(request, distribution_details[:quantity], distribution_details[:total_quantity])
+  end
+
+  def create_distribution_record(request:, bench:, pot:, details:)
+    create_request_distribution(
+      request: request,
+      bench: bench,
+      pot: pot,
+      plant_stage: request.plant_stage,
+      quantity: details[:quantity],
+      position: details[:position],
+      dimensions: details[:dimensions]
+    )
+  end
```

Comment:
pourquoi avoir défini une autre méthode qui va de toute façon réutiliser les mêmes clés ?
```suggestion
def create_distribution_record(request:, bench:, pot:, details:)
RequestDistribution.create!(
      request:,
      bench:,
      pot:,
      plant_stage: request.plant_stage,
      pot_quantity: details[:quantity],
      positions_on_bench: details[:position],
      dimensions: details[:dimensions]
    )
end
```
et avec ça, plus besoin de la méthode `create_request_distribution`